src/librustc_mir/borrow_check/nll/universal_regions.rs

```diff
@@ -490,6 +501,13 @@ impl<'cx, 'gcx, 'tcx> UniversalRegionsBuilder<'cx, 'gcx, 'tcx> {
             &bound_inputs_and_output,
             &mut indices,
         );
+
+        if self.mir_def_id == closure_base_def_id {
```

Comment:
Similarly, I would put a comment *here* like


---

Converse of above, if this is a *function* then the late-bound regions declared on its signature are local to the fn.